react-1/src/PresentationList.js

```diff
@@ -1,32 +1,58 @@
 // @flow
 
-import React, { Component } from 'react';
+import React, { Component, Fragment } from 'react';
 import ListItem, { type Attending } from './ListItem';
+import InputField from './InputField';
 import data from './data/agenda.json';
 
-type State = {
+type State = {|
   attending: Array<Attending>,
-};
+  filterString: string,
+|};
+
 class List extends Component<{||}, State> {
-  state = { attending: data.map((entry, idx) => false) };
+  state = {
+    attending: data.map((entry, idx) => false),
+    filterString: '',
+  };
 
-  handleChange(index: number, attending: Attending) {
+  handleAttendingChange(index: number, attending: Attending) {
     this.state.attending[index] = attending;
     this.setState({ attending: this.state.attending });
   }
 
+  handleFilterSearchChange(str: string) {
+    this.setState({ filterString: str.toLowerCase() });
+  }
+
   render() {
+    const { filterString } = this.state;
+    const filteredData = data.filter(
+      entry =>
+        entry.title.toLowerCase().includes(filterString) ||
+        entry.speakers.some(speaker =>
+          speaker.toLowerCase().includes(filterString)
+        )
+    );
```

Comment:
voir https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization sur ce sujet, on pourra peut-être le mettre dans un react-n

Reply:
Yep, très bon exemple